pkg/deviceshifu/deviceshifubase/deviceshifubase_config.go

```diff
@@ -212,3 +214,111 @@ func newEdgeDeviceRestClient(config *rest.Config) (*rest.RESTClient, error) {
 
 	return exampleRestClient, nil
 }
+
+// init DeviceShifuConfig With default
+func (dsConfig *DeviceShifuConfig) init() error {
```

Comment:
prefer to use shortest name for the function receiver.

dsConfig *DeviceShifuConfig -> dsc *DeviceShifuConfig
or
cfg *DeviceShifuConfig is pretty acceptable

Reply:
I think we should use short var names but remain readable. var names like dsc doesn't really tell you what this variable does, so I think dsConfig is suffice.

Reply:
normally, the function receiver can use one char. The philosophy of this shortest prefer is because the repeating cost. When the name is repeating a lot of times in a small scope I think the redundant problem is much more obvious we need to solve.

Reply:
To be honest I think one char receiver var name is just a very bad design which also reflects the famous notion: Premature optimization is the source of all evil. This philosophy of golang is just a bad as the error handling and no generic ideas. 
Readability should always comes before redundant, single char variable barely contains any information about what this var does and and what it is created for.

Reply:
if we have multiple single char in the code, I'd prefer not using that. eg c.a.b.func()
if the char is already in the current scope and it was declared during the context start.
 eg, (c *DeviceShifuConfig) func1 {c.func2()} I think that is a good practice.

Reply:
For the author of the code, `c.func2()` can be straightforward, but for future maintainers, `c.func2()` style can cause them a lot of trouble understanding what this code is doing. I myself have a lot of experience ,when viewing code that I didn't write, have to look back and forth what `c` is and a lot of times I didn't even realize `c` is the receiver of the function.

Reply:
My deep opinion is write more functional go code and try reduce write func with a receiver, I believe that's much OO style coding ha. the receiver is the global dependency and side effect of that block of code...

Reply:
Golang is not actually functional, it doesn't support partial function, continuation passing style, and higher order functions like fold. It call itself data oriented programming, which to me is a bad combination of oo and functional. 
And one thing bad about golang's receiver is that golang allow users to write receiver functions everywhere, and allow arbitrary names for receivers, which jeopardize readability badly. In many cases you can't really distinguish receivers from function parameters and local variables.
Despite how much I hate Rust, it did a much better job. Rust limited receiver functions inside a certain **impl** closure, and enforced receiver's name with the key word **self** which largely enhanced readibility.